server/webhook.go

```diff
@@ -161,41 +164,160 @@ func (wh *webhook) PostNotifications(p *Plugin, instanceID types.ID) ([]*model.P
 	return posts, http.StatusOK, nil
 }
 
-func newWebhook(jwh *JiraWebhook, eventType string, format string, args ...interface{}) *webhook {
-	return &webhook{
-		JiraWebhook: jwh,
-		eventTypes:  NewStringSet(eventType),
-		headline:    jwh.mdUser() + " " + fmt.Sprintf(format, args...) + " " + jwh.mdKeySummaryLink(),
+func (wh *webhook) checkIssueWatchers(p *Plugin, instanceID types.ID) {
+	var err error
+	var instance Instance
+	instance, err = p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		// This isn't an internal server error. There's just no instance installed.
+		p.errorf("error while loading instance, err: %v", err)
+		return
+	}
+
+	if instance == nil {
+		p.errorf("error : no instance found")
+		return
+	}
+
+	ci, ok := instance.(*cloudInstance)
+	if !ok {
+		p.errorf("error : no cloud instance found")
+		return
+	}
+
+	client, err := ci.getClientForBot()
+	// client, err := instance.GetClient(c)
+	if err != nil {
+		p.errorf("error while geting client for bot , err : %v", err)
+		return
+	}
+
+	watcherUsers, resp, err := client.Issue.GetWatchers(wh.Issue.Key)
```

Comment:
@hanzei @mickmister 

tried creating client and fetching watcher for the issue 

i get this error 
```
{"timestamp":"2022-02-23 14:50:10.192 Z","level":"error","msg":"error while geting watchers for issue , err : The 'accountId' query parameter needs to be provided: Request failed. Please analyze the request body for more details. Status code: 400: http2: response body closed: http2: response body closed","caller":"app/plugin_api.go:931","plugin_id":"jira"}
```

Reply:
Can you confirm that `connection.AccountID` in [connection.AccountID](https://github.com/demansoltech/mattermost-plugin-jira/blob/40692bf83915a8029ed819dc1ca81c99cd11cb75/server/client.go#L206) contains a non-zero value?

Reply:
@hanzei connection is fetched with help of jiraAccountID as from the webhook request body we can not assure that the id which we get is connected so we are using the function getClientForBot , can refer it [here](https://github.com/mattermost/mattermost-plugin-jira/blob/c6a6f006242e7d9f6dec71a3504a5d7bc08a35ad/server/instance_cloud.go#L207) and this client is created without accountID

let me know if i understood this wrong.

Reply:
@hanzei @sibasankarnayak If using the service account is causing issues (which I was thinking that it may), let's focus on the user token strategy instead. We need to support this on Jira Server anyway, and we don't have a service account to use there.

Reply:
@hanzei i see the same error for service account in other cases too.
@mickmister are you referring to add a user token in admin console to fetch the watching details ?

Reply:
> @mickmister are you referring to add a user token in admin console to fetch the watching details ?

No, I'm suggesting we look through the webhook event payload to find a connected user that would have access to this particular issue. Then we use that user's token from the kv store to fetch the watchers from the issue.

Reply:
@mickmister in this case there will be some situation where `author, reporter and assignee` none of the user will be connected . should we ignore those cases ?

Reply:
> @mickmister in this case there will be some situation where `author, reporter and assignee` none of the user will be connected . should we ignore those cases ?

@sibasankarnayak Yes, we agreed that we would ignore in that case